### PR TITLE
posix: Introduce getentropy() function

### PR DESCRIPTION
**Pull request by duda-patryk**
The function writes random data to the provided buffer. Maximum permitted buffer size is 256 bytes. On success the function returns 0, otherwise it returns -1 and sets errno to appropriate value.

### CODE REVIEW

File: tests/posix/getentropy/src/main.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+#include <zephyr/ztest.h>
+
+#include <unistd.h>
+
+ZTEST(getentropy_test_suite, test_getentropy_too_large)
+{
+	uint8_t buf[256 + 1] = { 0 };
+
+	int ret = getentropy(buf, sizeof(buf));
```
Comment (ycsin) [inline tests/posix/getentropy/src/main.c:15]: You will need to have 1 empty line after the end of var declaration block

Comment (duda-patryk) [reply tests/posix/getentropy/src/main.c]: Fixed. Thanks!

**Review by cfriedt — DISMISSED** [review]
Lgtm 👍

**Review by fabiobaltieri — DISMISSED** [review]

**Review by ycsin — DISMISSED** [review]

**Review by aescolar — CHANGES_REQUESTED** [review]
A minor thing

File: lib/posix/options/Kconfig.getentropy [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2024 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+DT_CHOSEN_ZEPHYR_ENTROPY := zephyr,entropy
+
+config GETENTROPY
+	bool "Support for getentropy"
+	depends on !NATIVE_APPLICATION
```
[the remark this answers is not in the document]
Comment (duda-patryk) [reply lib/posix/options/Kconfig.getentropy:9]: When I add `select NATIVE_LIBC_INCOMPATIBLE` I got some dependency loop.

[the remark this answers is not in the document]
Comment (duda-patryk) [reply lib/posix/options/Kconfig.getentropy:9]: The dependency loop: PINCTRL selected by I2C selected by FDC2X1X depends on FULL_LIBC_SUPPORTED selected by NATIVE_LIBC depends on EXTERNAL_LIBC depends on NATIVE_LIBC_INCOMPATIBLE depends on ENTROPY_HAS_DRIVER selected by ENTROPY_NEORV32_TRNG depends on SYSCON depends on PINCTRL.

I analyzed it, looks like PINCTRL_XLNX_ZYNQ has `select SYSCON` while it depends on it actually. I'll prepare patch for this.

[the remark this answers is not in the document]
Comment (duda-patryk) [reply lib/posix/options/Kconfig.getentropy:9]: Created #72022. I'll push version with `select NATIVE_LIBC_INCOMPATIBLE`, but expect build failure until #72022 is merged.

**Review by aescolar — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
hey CI is sad `2024-04-26T17:58:18.6104707Z /__w/zephyr/zephyr/scripts/kconfig/kconfig.py: Dependency loop`

**Comment by duda-patryk** [comment]
> hey CI is sad `2024-04-26T17:58:18.6104707Z /__w/zephyr/zephyr/scripts/kconfig/kconfig.py: Dependency loop` https://productionresultssa0.blob.core.windows.net/actions-results/d8e18830-7f04-4e7d-b8d1-82e77f3797d2/workflow-job-run-722c1006-548e-57d0-0167-cec5b9d9ca3c/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-04-29T08%3A58%3A16Z&sig=HOtQ6IlDu69WBxWua0gkChZfkDBixFjObmHqTFFpsTY%3D&sp=r&spr=https&sr=b&st=2024-04-29T08%3A48%3A11Z&sv=2021-12-02

Fix https://github.com/zephyrproject-rtos/zephyr/pull/72022. Unfortunately, it needs to wait 34 hours, for changing 3 lines

**Comment by aescolar** [comment]
@fabiobaltieri yep, https://github.com/zephyrproject-rtos/zephyr/pull/72022 needs merging first

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri yep, #72022 needs merging first

That one looks simple enough to me, I tagged it as trival and merged it, going to kick CI here again.

**Review by cfriedt — APPROVED** [review]

**Pull request merged**